include/seqan3/core/detail/debug_stream_range.hpp

```diff
@@ -17,6 +17,52 @@
 #include <seqan3/core/type_traits/range.hpp>
 #include <seqan3/std/ranges>
 
+namespace seqan3::detail
+{
+/*!\brief A helper concept definition for ranges that can be streamed to the seqan3::debug_stream.
+ * \tparam rng_t The range type to check.
+ * \ingroup core
+ *
+ * \details
+ *
+ * This concept refines the std::ranges::InputRange concept to allow streaming the range object to the debug stream,
+ * with the following requirements:
+ *
+ * * `rng_t` is not the same type as `reference_t<rng_t>`,
+ * * `rng_t` is not a pointer or c-style array,
+ * * `reference_t<rng_t>` is not `char`.
+ *
+ * \note If you need to further refine this concept for your type ensure that this concept comes before any
+ *       additional refinement, otherwise your concept definition/requires clause is ill-formed according to the
+ *       subsumption rules for concepts.
```

Comment:
can you give a code example what you mean with this?

Reply:
😭 Ok

Reply:
I did remove it because it is not valid. The order of the constraints is not relevant as long as they are atomically the same (and refined of course)